src/core/border_router/routing_manager.cpp

```diff
@@ -449,7 +449,11 @@ void RoutingManager::EvaluateOmrPrefix(OmrPrefixArray &aNewOmrPrefixes)
         // The `aNewOmrPrefixes` remains empty if we fail to publish
         // the local OMR prefix.
     }
-    else if (publishedLocalOmrPrefix != nullptr && smallestOmrPrefix != publishedLocalOmrPrefix)
+    else if (*smallestOmrPrefix == mLocalOmrPrefix)
```

Comment:
@abtink @wgtdkp 
This condition handles the case where `mLocalOmrPrefix` is already in Leader's Network Data, but not locally added. It is possible when the BR restores the Network Data from Leader. In such case the BR should re-add `mLocalOmrPrefix` to local Network Data so that it keeps advertising OMR Prefix. 
Otherwise, the OMR Prefix will be removed from Leader's Network Data because of Network Data inconsistentency, and then all BRs start to re-elect OMR Prefix. So, the change will make the OMR Prefix more stable in some corner cases.

Reply:
Perfect. Thanks. 👍 

Can you please also add this (or summary of this) in the commit message?

It's alway a good practice to add a short description of the change/fix when the commit/change is larger (non-trival). The commit logs remain with code and help others later looking at a history (will be appreciated by others looking at the code in future). 😃

Reply:
+1

Reply:
done